core-plugins/src/main/java/io/cdap/plugin/batch/action/EmailAction.java

```diff
@@ -93,7 +93,9 @@ public PasswordAuthentication getPasswordAuthentication() {
       javaMailProperties.put("mail.smtp.ssl.enable", true);
     }
     if ("TLS".equalsIgnoreCase(config.protocol)) {
-      javaMailProperties.put("mail.smtp.starttls.enable", true);
+      javaMailProperties.put("mail.smtp.starttls.enable", "true");
+      javaMailProperties.put("mail.smtp.ssl.trust", config.host);
+      config.protocol = "smtp"; // protocol is smtp not tls.
```

Comment:
The `config.protocol` field directly represents the text input for the `protocol` field displayed in the UI, so it might be a bit jarring to change it from under the user. Can we instead pass in a separate variable to line 123?

Reply:
yeah I agree